lib/compat/wordpress-6.5/navigation-block-variations.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * Workaround for Core versions < 6.5 to register navigation link variations on registration
+ *
+ * @param array $args The block type arguments.
+ * @return array The updated block type arguments.
+ */
+function gutenberg_navigation_link_variations_compat( $args ) {
+
+	if ( 'core/navigation-link' !== $args['name'] || ! empty( $args['variation_callback'] ) ) {
+		return $args;
+	}
+	$args['variation_callback'] = 'build_navigation_link_block_variations';
```

Comment:
This compat code doesn't work because the `build_navigation_link_block_variations` function does not exist.

The consequence is that in WordPress 6.4 the Navigation Link variations for post types (Post Link, Page Link) are not registered, we have only "Custom Link" available in the editor.

I was seeing strange Playwright e2e failures in my PR #59064 and I tracked them down to this. The e2e tests for the `core/navigation` block are failing because they expect "Page Link" components, but they see only "Custom Link" ones. See for example this code:

https://github.com/WordPress/gutenberg/blob/237865fad0864c209a7c3e771e23fe66f4fbca25/test/e2e/specs/editor/blocks/navigation-list-view.spec.js#L103-L112

Reply:
Oh gosh, this PR (or my work on it) seems to be cursed 🙈 
You are right, the functions name should be `gutenberg_block_core_navigation_link_build_variations` - since that's the prefixed name after building. 
If I change it to that and use WP 6.4 locally, I get "Page Link" and "Post Link" variations correctly. 
I wonder why no other tests caught that, but since the function is just passed as a callback it's not called directly and therefore doesn't trigger an error - probably that's why.

I've created a PR to fix that here: https://github.com/WordPress/gutenberg/pull/59126